02-peer-protocol.md

```diff
@@ -560,6 +1072,342 @@ ones it has sent so it can use them should they be requested by
+      channel's funding output
+
+##### Rationale
+
+The channel funding output must be added by the *opener*, who pays its fees.
+
+#### The `tx_complete` Message
+
+Upon receipt of consecutive `tx_complete`s, the receiving node:
+  - if is the *accepter*:
+    - MUST fail the negotiation if:
+      - no funding output was received
+      - the value of the funding output is not equal to the sum of
+        `open_channel2`.`funding_satoshis` and `accept_channel2`.
+        `funding_satoshis`
+      - the value of the funding output is less than the `dust_limit`
+  - if it is an RBF attempt:
+    - MUST fail the negotiation if:
+      - the transaction's total fees is less than the last
+        successfully negotiated transaction's fees
+      - the transaction does not share a common input with all previous
+        funding transactions
+  - if it has sent `require_confirmed_inputs` in `open_channel2` or `accept_channel2`:
+    - MUST fail the negotiation if:
+      - one of the inputs added by the other peer is unconfirmed
+
+### The `commitment_signed` Message
+
+This message is exchanged by both peers. It contains the signatures for
+the first commitment transaction.
+
+Rationale and Requirements are the same as listed below,
+for [`commitment_signed`](#commiting-updates-so-far-commitment_signed) with the following additions.
+
+#### Requirements
+
+The sending node:
+  - MUST send zero HTLCs.
+  - MUST remember the details of this funding transaction.
+
+The receiving node:
+  - if the message has one or more HTLCs:
+    - MUST fail the negotiation
+  - if it has not already transmitted its `commitment_signed`:
+    - MUST send `commitment_signed`
+  - Otherwise:
+    - MUST send `tx_signatures` if it should sign first, as specified
+      in the [`tx_signatures` requirements](#the-tx_signatures-message)
+
+#### Rationale
+
+The first commitment transaction has no HTLCs.
+
+Once peers are ready to exchange commitment signatures, they must remember
+the details of the funding transaction to allow resuming the signatures
+exchange if a disconnection happens.
+
+### Sharing funding signatures: `tx_signatures`
+
+After a valid `commitment_signed` has been received
+from the peer and a `commitment_signed` has been sent, a peer:
+  - MUST transmit `tx_signatures` with their signatures for the funding
+    transaction, following the order specified in the
+    [`tx_signatures` requirements](#the-tx_signatures-message)
+
+#### Requirements
+
+The sending node:
+  - MUST verify it has received a valid commitment signature from its peer
+  - MUST remember the details of this funding transaction
+  - if it has NOT received a valid `commitment_signed` message:
+    - MUST NOT send a `tx_signatures` message
+
+The receiving node:
+  - if the `witness` weight lowers the effective `feerate`
+    below the the *opener*'s feerate for the funding transaction:
+    - SHOULD broadcast their commitment transaction, closing the channel.
```

Comment:
You're correct, the channel can still operate normally if this is violated.

The rationale describes why this is suggested: "This should disincentivize peers from underpaying fees."

Note that it's at the level of recommendation (SHOULD) not directive (MUST). I updated the ordering + some of the wording for this in an attempt to make it clearer.

Reply:
But does it really disincentivize the behavior?  Force closing the channel may very well hurt us more than the peer.  If we're the opener, we will likely pay significantly more fees than the peer -- more fees on the funding tx and all the fees on the commitment tx.  If we're the accepter, maybe there's a disincentive, or maybe not -- it depends on the fees paid by us on the funding tx.

I think the best course of action in this situation is to double spend productively if there's an opportunity.  Otherwise, we can just wait -- we don't have another use for those funds anyway.

And we need to have this double-spending logic in place anyway to handle griefing from withholding `tx_signatures`.  Why not use it to address this griefing vector as well?

Reply:
Reasonably, if you don't believe that the feerate on a channel open is good enough to be mined, you're safe to broadcast the commitment signatures and double-spend your inputs to the channel.

This covers the case where the existing channel gets mined while you're waiting for the other use of those inputs to be mined/happen.

In the case that the channel does eventually open, then it'll immediately close. If it never opens, then you never spend the onchain fees.